src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -743,6 +756,17 @@ static void rr_update_locked(grpc_exec_ctx *exec_ctx, grpc_lb_policy *policy,
       gpr_zalloc(sizeof(subchannel_data) * num_addrs);
   subchannel_list->num_subchannels = num_addrs;
   gpr_ref_init(&subchannel_list->refcount, 1);
+  if (p->latest_pending_subchannel_list != NULL && p->started_picking) {
+    if (GRPC_TRACER_ON(grpc_lb_round_robin_trace)) {
+      gpr_log(GPR_DEBUG,
+              "Shutting down latest pending subchannel list %p, about to be "
+              "replaced by newer latest %p",
+              (void *)p->latest_pending_subchannel_list,
+              (void *)subchannel_list);
+    }
+    rr_subchannel_list_shutdown(exec_ctx, p->latest_pending_subchannel_list,
```

Comment:
As per our discussion, I think there are duplicate refs here for the ones you identified as type 1.  I think the subchannel list does not need to hold a reference to itself -- the one returned from `rr_subchannel_list_create()` is the one that should be held by the RR policy, so there's no need for an additional ref on line 791 below.  And then there is also no need for an additional unref every time we shut down a subchannel list.  So we can rename `rr_subchannel_list_shutdown()` to `rr_subchannel_list_shutdown_and_unref()`.